### PR TITLE
Feat/optimize the commands calls zappy gui

### PR DESCRIPTION
**Pull request by Thomaltarix**
With the help of @mathieurobert1, i setup the a clock for each player last `ppo` command call.
The `ppo` command is used to update the player's position.
Now, if the last ppo call last from more that 50 ticks, the Gui is sending `ppo` for this player to the server.
This method avoid sending `ppo` to the server for each tick for nothing.

### CODE REVIEW

File: gui/src/Engine/Engine.cpp [hunk 1]
```diff
@@ -71,7 +71,13 @@ void Gui::Engine::sendMessageUpdate()
     _network.get()->sendMessageServer("sgt\n");
     for (auto &team : _gameData.get()->getTeams()) {
         for (auto &player : team.getPlayers()) {
-            _network.get()->sendMessageServer("ppo " + std::to_string(player.getId()) + "\n");
+            Player::PlayerState state = player.getState();
+            if (state == Player::PlayerState::DEAD || state == Player::PlayerState::INCANTATION)
+                continue;
+            if ((float)(currentTick - player.getLastPositionUpdate()) / CLOCKS_PER_SEC > 50 / _gameData->getServerTick()) {
```
Comment (Marius-P1) [inline gui/src/Engine/Engine.cpp:77]: if CLOCKS_PER_SEC > 50 = False then it's replaced by 0 so the operation is 
(float)(currentTick - player.getLastPositionUpdate()) / 0 / _gameData->getServerTick()
Don't that make a division by 0 ?

**Review by Marius-P1 — COMMENTED** [review]
Just a small question before i approve

**Review by mathieurobert1 — APPROVED** [review]
Perfect, just wait before the @AxelF44 's pr about sending ppo when need it.

File: gui/src/Engine/Engine.cpp (hunk 1 above)
Comment (Thomaltarix) [reply gui/src/Engine/Engine.cpp:77]: ??
x)
That's:
``` 
(float)(currentTick -player.getLastPositionUpdate()) / CLOCKS_PER_SEC // first operand
>
50 / _gameData->getServerTick() // second operand
```

Comment (Marius-P1) [reply gui/src/Engine/Engine.cpp:77]: Lack of understanding on my part, mb

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Pull request closed without merging**